### PR TITLE
Implement more stack operations

### PR DESCRIPTION
**Pull request by tnachen**

### CODE REVIEW

File: assembly/src/v1/parsers/stack_ops.rs [hunk 1]
```diff
@@ -3,39 +3,141 @@ use super::{AssemblyError, Operation, Token};
...
+/// the index of the word.
+pub fn parse_dupw(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => {
+            span_ops.push(Operation::Dup3);
+            span_ops.push(Operation::Dup2);
+            span_ops.push(Operation::Dup1);
+            span_ops.push(Operation::Dup0);
+        }
```
Comment (bobbinth) [inline assembly/src/v1/parsers/stack_ops.rs:87]: Shouldn't this be: `DUP3 DUP3 DUP3 DUP3`?

File: assembly/src/v1/parsers/stack_ops.rs [hunk 2]
```diff
@@ -3,39 +3,141 @@ use super::{AssemblyError, Operation, Token};
...
+                span_ops.push(Operation::Dup3);
+                span_ops.push(Operation::Dup2);
+                span_ops.push(Operation::Dup1);
+                span_ops.push(Operation::Dup0);
+            }
+            "1" => {
+                span_ops.push(Operation::Dup7);
+                span_ops.push(Operation::Dup6);
+                span_ops.push(Operation::Dup5);
+                span_ops.push(Operation::Dup4);
```
Comment (bobbinth) [inline assembly/src/v1/parsers/stack_ops.rs:99]: Shouldn't this be: `DUP7 DUP7 DUP7 DUP7`?

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks great! Thank you! I've left a couple of comments about `dupw.n` parser - but other than that, everything looks good.

File: assembly/src/v1/parsers/stack_ops.rs (hunk 2 above)
Comment (tnachen) [reply assembly/src/v1/parsers/stack_ops.rs]: ah yes you're right! Fixed all the other places

**Review by bobbinth — APPROVED** [review]
Everything looks great! Thank you!

**Pull request merged**